src/main/java/moli/Moli.java

```diff
@@ -0,0 +1,196 @@
+package moli;
+
+import moli.task.Task;
+import moli.task.Todo;
+import moli.task.Deadline;
+import moli.task.Event;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Moli {
+    private static final int MAX_TASKS = 100;
+    private static final ArrayList<Task> tasks = Storage.loadTasks(); // Load tasks from file
+    private static final String DIVIDER = "____________________________________________________________";
+
+    public static void main(String[] args) {
+        final String LOGO = "\n"
+                + "                ███╗   ███╗ ██████╗ ██╗     ██╗\n"
+                + "                ████╗ ████║██╔═══██╗██║     ██║\n"
+                + "                ██╔████╔██║██║   ██║██║     ██║\n"
+                + "                ██║╚██╔╝██║██║   ██║██║     ██║\n"
+                + "                ██║ ╚═╝ ██║╚██████╔╝███████╗██║\n"
+                + "                ╚═╝     ╚═╝ ╚═════╝ ╚══════╝╚═╝";
+
+        System.out.println(LOGO);
+        System.out.println(DIVIDER);
+        System.out.println("Hello, dear one. I'm Moli.\nI'm here for you💙");
+        System.out.println(DIVIDER);
+
+        Scanner scanner = new Scanner(System.in);
+        boolean hasQuit = false;
+        while (!hasQuit) {
+            String input = scanner.nextLine().trim();
+            try {
+                if ("bye".equalsIgnoreCase(input)) {
+                    System.out.println(DIVIDER);
+                    System.out.println("Take care of yourself💖\nCome back anytime you need a listening ear.");
+                    System.out.println(DIVIDER);
+                    hasQuit = true;
+                } else if ("list".equalsIgnoreCase(input)) {
+                    showTaskList();
+                } else if (input.startsWith("mark ")) {
+                    markTask(input);
+                } else if (input.startsWith("unmark ")) {
+                    unmarkTask(input);
+                } else if (input.startsWith("delete ")) {
+                    deleteTask(input);
+                } else if (input.startsWith("todo")) {
+                    if (input.length() <= 4) {
+                        throw new MoliException("Sorry. A todo task must have a description, I'm here to listen.");
+                    }
+                    addTask(new Todo(input.substring(5).trim()));
+                } else if (input.startsWith("deadline")) {
+                    if (!input.matches("^deadline .+ /by .+$")) {
+                        throw new MoliException("Sorry, please enter the correct format: 'deadline TASK_DESCRIPTION /by DEADLINE_DATE'");
+                    }
+                    String[] parts = input.substring(9).split(" /by ", 2);
+                    addTask(new Deadline(parts[0].trim(), parts[1].trim()));
+                } else if (input.startsWith("event")) {
+                    if (!input.matches("^event .+ /from .+ /to .+$")) {
```

Comment:
Try and avoid the usage of magic literals by storing them in constants or variables. If it is not practical to do so, it would help the reader if you add concise comments explaining the parsing happening here